internal/controller/utils.go

```diff
@@ -45,3 +47,36 @@ func getUsedClusterClasses(ctx context.Context, c client.Client, namespace strin
 
 	return usedClusterClasses, nil
 }
+
+func helmTemplate(restConfig *rest.Config, chartPath, releaseName, namespace string) ([]byte, error) {
```

Comment:
the function helmTemplate() contains only one line, and the function is used exactly once. Please inline the function. Or do you see good reasons to keep the method?

Reply:
@NotTheEvilOne to be able to test your branch, I created a new branch with the changes from main merged into it. I will create a new PR containing your changes. I will inline this method in this PR, so you don't need to. ok?

Reply:
I tend to leave this function as it is (especially if it may be required for other purposes in the future) as it improves readability of the code:

`template, err := helmTemplate(r.RESTConfig, clusterClassChart, releaseName, namespace)`
vs.
`template, err := helmTemplateWithValues(r.RESTConfig, clusterClassChart, releaseName, namespace, "")`
(It would be even uglier if we would accept `nil` here.

The branch has already been rebased and should be mergable without intervention.

Reply:
I also just had a look. I agree with you @NotTheEvilOne. However, in the other controller you do use helmTemplateWithValues directly, even though the same applies and two out of the 5 values are hard-coded. Can you also add a wrapper function there to make it more readable?

Reply:
AFAIK both values are more suitable to be replaced by constants (one already is). I don't think it makes sense to add a separate wrapper here as at least `clusterAddonNamespace` is used in multiple cases and `"cluster-addon"` is used both in `clusteraddon_controller.go` and `clusterstackrelease_controller.go`. The wrapper for that instance is less likely to be reused in a different place and should be "inlined" therefore.

Reply:
I'm talking about readability.. Having a function with 5 inputs is less readable than 2-3 inputs. That's why I support the fact that we should have a wrapper function in both cases. Otherwise I agree with @guettli and would remove all wrappers, if this is not about readability

Reply:
Added as requested.